### ISSUE TITLE
lastNotification/timesSent get lost upon update subscription in noCache cases

### ISSUE THREAD

**Issue by fgalan** [telefonicaid/fiware-orion#3429]
At https://github.com/telefonicaid/fiware-orion/blob/master/test/functionalTest/cases/3189_oneshot_subscription/3189_oneshot_subscription_noCache.test we get a subscription just after updating it (to change status inactive -> oneshot).

```
10. GET /v2/subscriptions/subscription_id, to check the update status.
======================================================================
HTTP/1.1 200 OK
Content-Length: 371
Content-Type: application/json
Fiware-Correlator: REGEX([0-9a-f\-]{36})
Date: REGEX(.*)

{
    "description": "A subscription to get info about Room1", 
    "id": "REGEX([0-9a-f]{24})", 
    "notification": {
        "attrs": [
            "temperature"
        ], 
        "attrsFormat": "normalized", 
        "http": {
            "url": "http://localhost:REGEX(\d+)/notify"
        }, 
        "lastSuccess": "REGEX(.*)",
        "lastSucessCode": 200
    }, 
    "status": "oneshot", 
    "subject": {
        "condition": {
            "attrs": [
                "pressure"
            ]
        }, 
        "entities": [
            {
                "id": "Room1", 
                "type": "Room"
            }
        ]
    }
}
```

Having a `lastSuccess` without `lastNotification` doesn't make sense. It is like `lastNotification` and `timesSent` got lost after updating the subscription with PATCH.

Note this is a noCache test. If we look to https://github.com/telefonicaid/fiware-orion/blob/master/test/functionalTest/cases/3189_oneshot_subscription/3189_oneshot_subscription.test, the same steps seems to be ok.

So maybe the bug is scoped to noCache update subscriptions scenarios. Some possiblities:

1. It affects to all subscription update cases
2. It affects to subscription update involving status
3. It affect to subscription updte involving status but only when new status is "onsehot"

Needs to be debugged.

CC: @SourabhChourasiya-NEC

**Comment by ArqamFarooqui110719** [comment]
Hi @fgalan sir, Whether this issue is still valid?

**Comment by ArqamFarooqui110719** [comment]
As per my investigation, When running the Orion in `-nocache`  mode as below:
```
[root@localhost fiware-orion]# ps -aux|grep contextBroker
root      104325  0.0  0.8 541452 15368 ?        Ssl  00:03   0:00 contextBroker -logLevel DEBUG -t 0-255 -noCache
root      113914  0.0  0.0  12140  1176 pts/1    S+   00:17   0:00 grep --color=auto contextBroker
```

I'm getting below response after updating the subscription status from inactive to oneshot:

Updated the subscription:
```
[root@localhost fiware-orion]# curl localhost:1026/v2/subscriptions/62167ea63b22d4616a58896b -s -S -X PATCH -H 'Content-Type: application/json' -d @- <<EOF
{
  "status": "oneshot"
}
EOF

[root@localhost fiware-orion]#
```

Get the subscription:
<pre>
[root@localhost fiware-orion]# curl -v localhost:1026/v2/subscriptions/62167ea63b22d4616a58896b -s -S  -H 'Accept: application/json' | python3 -mjson.tool
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 1026 (#0)
> GET /v2/subscriptions/62167ea63b22d4616a58896b HTTP/1.1
> Host: localhost:1026
> User-Agent: curl/7.61.1
> Accept: application/json
>
< HTTP/1.1 200 OK
< Connection: Keep-Alive
< Content-Length: 524
< Content-Type: application/json
< Fiware-Correlator: ae84599a-94d8-11ec-8ac9-080027882e60
< Date: Wed, 23 Feb 2022 18:44:50 GMT
<
{ [524 bytes data]
* Connection #0 to host localhost left intact
{
    "id": "62167ea63b22d4616a58896b",
    "description": "A subscription to get info about Room2",
    "expires": "2040-01-01T14:00:00.000Z",
    "status": "oneshot",
    "subject": {
        "entities": [
            {
                "id": "Room2",
                "type": "Room"
            }
        ],
        "condition": {
            "attrs": [
                "pressure",
                "temperature"
            ]
        }
    },
    "notification": {
        "timesSent": 1,
        "lastNotification": "2022-02-23T18:39:21.000Z",
        "attrs": [
            "temperature",
            "pressure"
        ],
        "onlyChangedAttrs": false,
        "attrsFormat": "normalized",
        "http": {
            "url": "http://localhost:1028/accumulate"
        },
        "lastSuccess": "2022-02-23T18:39:21.000Z",
        "lastSuccessCode": 200
    }
}
</pre>



What is your opinion/suggestion on this?
Thanks:)

**Comment by fgalan** [comment]
The https://github.com/telefonicaid/fiware-orion/blob/master/test/functionalTest/cases/3189_oneshot_subscription/3189_oneshot_subscription_noCache.test test was recently modified at commits:

* https://github.com/telefonicaid/fiware-orion/commit/831ff8e43b8a3e48165661c73c7e2c638d739f34, belonging to PR #3983
* https://github.com/telefonicaid/fiware-orion/commit/4a7d490cfceafbea13ecabd3ed74d89bd9f5a855, belonging to PR #3982

Both of them were merged to Orion for version 3.4.0.

So, it seems that the issue was solved at Orion 3.4.0. The above modifications in .test, re-introducing `timesSent` and `lastNotification` pro

Thus, this issues is solved and can be closed. Thanks for noticing!
